grakn-engine/src/main/java/ai/grakn/engine/controller/GraqlController.java

```diff
@@ -112,25 +107,14 @@ private Object executeGraql(Request request, Response response) {
         String keyspace = mandatoryQueryParameter(request, KEYSPACE);
         boolean infer = parseBoolean(mandatoryQueryParameter(request, INFER));
         boolean materialise = parseBoolean(mandatoryQueryParameter(request, MATERIALISE));
+        int limitEmbedded = queryParameter(request, LIMIT_EMBEDDED).map(Integer::parseInt).orElse(-1);
         String acceptType = getAcceptType(request);
+
         try(GraknGraph graph = factory.getGraph(keyspace, WRITE)){
             Query<?> query = graph.graql().materialise(materialise).infer(infer).parse(queryString);
-            if(!validContentType(acceptType, query)) {
-                throw GraknServerException.contentTypeQueryMismatch(acceptType, query);
-            }            
-            if (query instanceof DeleteQuery) {
-                query.execute();
-                graph.commit();
-                return respond(response, APPLICATION_TEXT, Json.object());
-            }
-            else if (query instanceof InsertQuery) {
-                Object resp = respond(response, APPLICATION_JSON, executeQuery(request, query, acceptType));
-                graph.commit();
-                return resp;
-            }
-            else {
-                return respond(response, acceptType, executeQuery(request, query, acceptType));
-            }
+            Object resp = respond(response, acceptType, executeQuery(keyspace, limitEmbedded, query, acceptType));
+            graph.commit();
```

Comment:
Need to double check that this does not add overhead when just doing match queries